pkg/cmd/adm/install_operator.go

```diff
@@ -0,0 +1,200 @@
+		},
+		Spec: olmv1alpha1.CatalogSourceSpec{
+			SourceType:  olmv1alpha1.SourceTypeGrpc,
+			Image:       fmt.Sprintf("quay.io/codeready-toolchain/%s-operator-index:latest", operator),
+			DisplayName: "Dev Sandbox Operators",
+			Publisher:   "Red Hat",
+			UpdateStrategy: &olmv1alpha1.UpdateStrategy{
+				RegistryPoll: &olmv1alpha1.RegistryPoll{
+					Interval: &metav1.Duration{
+						Duration: 1 * time.Minute,
+					},
+				},
+			},
+		},
+	}
+}
+
+func newOperatorGroup(name types.NamespacedName) *olmv1.OperatorGroup {
+	return &olmv1.OperatorGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: name.Namespace,
+			Name:      name.Name,
+		},
+		Spec: olmv1.OperatorGroupSpec{
+			TargetNamespaces: []string{
+				name.Namespace,
+			},
+		},
+	}
+}
+
+func newSubscription(name types.NamespacedName, operatorName, catalogSourceName string) *olmv1alpha1.Subscription {
+	return &olmv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: name.Namespace,
+			Name:      name.Name,
+		},
+		Spec: &olmv1alpha1.SubscriptionSpec{
+			Channel:                "staging",
+			InstallPlanApproval:    olmv1alpha1.ApprovalAutomatic,
+			Package:                operatorName,
+			CatalogSource:          catalogSourceName,
+			CatalogSourceNamespace: name.Namespace,
+		},
+	}
+}
+
+func newKubeClient(ctx *extendedCommandContext, kubeConfigPath string) (cl runtimeclient.Client, err error) {
+	var kubeConfig *clientcmdapi.Config
+	var clientConfig *rest.Config
+
+	kubeConfig, err = clientcmd.LoadFromFile(kubeConfigPath)
+	if err != nil {
+		return
+	}
+	clientConfig, err = clientcmd.NewDefaultClientConfig(*kubeConfig, nil).ClientConfig()
+	if err != nil {
+		return
+	}
+	cl, err = ctx.NewClientFromRestConfig(clientConfig)
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func waitUntilCatalogSourceIsReady(ctx *clicontext.CommandContext, cl runtimeclient.Client, catalogSourceKey runtimeclient.ObjectKey, waitForReadyTimeout time.Duration) error {
+	return wait.PollImmediate(2*time.Second, waitForReadyTimeout, func() (bool, error) {
+		ctx.Printlnf("waiting for CatalogSource %s to become ready", catalogSourceKey)
+		cs := &olmv1alpha1.CatalogSource{}
+		if err := cl.Get(ctx, catalogSourceKey, cs); err != nil {
+			return false, err
+		}
+
+		return cs.Status.GRPCConnectionState != nil && cs.Status.GRPCConnectionState.LastObservedState == "READY", nil
+	})
+}
+
+func waitUntilInstallPlanIsComplete(ctx *clicontext.CommandContext, cl runtimeclient.Client, namespace string, waitForReadyTimeout time.Duration) error {
+	return wait.PollImmediate(2*time.Second, waitForReadyTimeout, func() (bool, error) {
+		ctx.Printlnf("waiting for InstallPlans in namespace %s to complete", namespace)
+		plans := &olmv1alpha1.InstallPlanList{}
+		if err := cl.List(ctx, plans, runtimeclient.InNamespace(namespace)); err != nil {
+			return false, err
+		}
+
+		for _, ip := range plans.Items {
+			if ip.Status.Phase != olmv1alpha1.InstallPlanPhaseComplete {
```

Comment:
Ok, makes sense. I saw that this is what the wait-until-is-installed.sh script does, but maybe I've missed something. Thanks for the suggestion. 

Done in https://github.com/kubesaw/ksctl/pull/48/commits/2c852fe2e05acba8b41d47d9d54bdbf647eb1d06

Reply:
the CI scripts can rely on the fact that the namespaces are managed by the logic/scripts themself. In other words, we don't expect that the namespace would contain anything else and we can assume that it's always a fresh new install. 
This command can be used in different situations, so that's why I would make it more robust.